src/blockdata/transaction.rs

```diff
@@ -332,19 +325,50 @@ impl Transaction {
     /// - Does NOT attempt to support OP_CODESEPARATOR. In general this would require evaluating
     /// `script_pubkey` to determine which separators get evaluated and which don't, which we don't
     /// have the information to determine.
-    /// - Does NOT handle the sighash single bug, you should either handle that manually or use
-    /// [`Self::signature_hash()`] instead.
+    /// - Does NOT handle the sighash single bug (see "`Option` return value" section)
+    ///
+    /// # `Option` return value
+    ///
+    /// This function can't handle the SIGHASH_SINGLE bug internally. However if it was
+    /// to return an error in this case, a lot of users would mishandle it by calling
+    /// `.expect()` on the result, assuming that `writer` they have passed can't possibly
+    /// fail unaware of this additional case. For this reason the `Result` is wrapped
+    /// in an `Option`. In the case of SIGHASH_SIGNLE bug, the function will return
+    /// `None` without writting anything to the `writer`.
```

Comment:
In 9435eda1077743065a9e22b031d21275a5ab7f34:

I think it's fine here to just say "we return `None` in the case of the `SIGHASH_SINGLE` bug, reserving the error return of the `Result` for writer errors, allowing users to `unwrap` when they have infallible writers". Also SIGHASH_SINGLE is misspelled once in this block.